astropy_healpix/core.py

```diff
@@ -116,12 +117,72 @@ def nside_to_pixel_resolution(nside):
     -------
     resolution : :class:`~astropy.units.Quantity`
         The resolution of the HEALPix pixels
+
+    See also
+    --------
+    pixel_resolution_to_nside
     """
     nside = np.asanyarray(nside, dtype=np.int64)
     _validate_nside(nside)
     return (nside_to_pixel_area(nside) ** 0.5).to(u.arcmin)
 
 
+def pixel_resolution_to_nside(resolution, round='nearest'):
+    """Find closest HEALPix nside for a given angular resolution.
+
+    This function is the inverse of `nside_to_pixel_resolution`,
+    for the default rounding scheme of ``round='nearest'``.
+
+    If you choose ``round='up'``, you'll get HEALPix pixels that
+    have at least the requested resolution (usually a bit better
+    due to rounding).
+
+    Pixel resolution is defined as square root of pixel area.
+
+    Parameters
+    ----------
+    resolution : `~astropy.units.Quantity`
+        Angular resolution
+    round : {'up', 'nearest', 'down'}
+        Which way to round
+
+    Returns
+    -------
+    nside : int
+        The number of pixels on the side of one of the 12 'top-level' HEALPix tiles.
+        Always a power of 2.
+
+    Examples
+    --------
+    >>> from astropy_healpix import pixel_resolution_to_nside
+    >>> pixel_resolution_to_nside('13 arcmin')
+    256
+    >>> pixel_resolution_to_nside('13 arcmin', round='up')
+    512
+    """
+    resolution = Angle(resolution).radian
```

Comment:
Currently you say above that ``resolution`` should be a Quantity, but in the example/tests you use a string and you parse it here into an ``Angle`` object. While this can be convenient, I think at least in the Examples we should encourage people to use Quantity objects - elsewhere in astropy-healpix so far we just assume the input is a Quantity and will fail with a string.

Just to be clear, I could be convinced to allow strings as inputs but we should then update the Parameters section to mention this as an option, and probably show both string and Quantity in the Examples (and we should maybe do the same for other functions).

Finally, I think you can probably add ``copy=False`` to the Angle call to avoid an unnecessary copy when a Quantity is passed in.

Reply:
Agreed that we should be consistent in our API and fully correct in the docstrings.
I've changed to the pattern you're using everywhere else for now, to require quantities to be passed in.

Adding support for strings for angle inputs as a convenience can be discussed in the future, and if we decide to add it, it should be done everywhere or it'll be confusing to users.